torch_geometric/nn/aggr/multi.py

```diff
@@ -27,21 +55,65 @@ def __init__(self, aggrs: List[Union[Aggregation, str]],
                              f"'{self.__class__.__name__}' "
                              f"(got '{len(aggrs_kwargs)}', "
                              f"expected '{len(aggrs)}'). Ensure that both "
-                             f"'aggrs' and 'aggrs_kwargs' are consistent")
+                             f"'aggrs' and 'aggrs_kwargs' are consistent.")
 
         self.aggrs = torch.nn.ModuleList([
             aggregation_resolver(aggr, **aggr_kwargs)
             for aggr, aggr_kwargs in zip(aggrs, aggrs_kwargs)
         ])
 
+        self.combine_mode = combine_mode
+        if combine_mode == 'proj':
+            if in_channels is None:
+                raise ValueError(
+                    f"'Combine mode '{combine_mode}' must "
+                    f"have `in_channels` specified (got '{in_channels}').")
+            if out_channels is None:
+                out_channels = in_channels
+            self.combine_lin = Linear(in_channels * len(aggrs), out_channels,
+                                      bias=True)
+        else:
+            if (in_channels or out_channels) is not None:
+                raise ValueError("Channel projection is only supported in "
+                                 "the `'proj'` combine mode.")
+            if combine_mode != "cat":
+                self.combine_aggr = aggregation_resolver(combine_mode)
```

Comment:
As said earlier, this will lead to an inefficient implementation.

Reply:
Good point.., I wonder if in the aggregation we could fall back to dense aggregation in the case that index is not provided. Does that make sense, WDTY?

Reply:
Sure, that is possible. We could first check for dense implementation, and for others fallback to sparse ones. My proposal would be to drop the support for sparse aggregation for now until we find a use-case or need to do so.

Reply:
Make sense! Use dense reduce now.